src/box/applier.cc

```diff
@@ -546,7 +548,6 @@ applier_fetch_snapshot(struct applier *applier)
 	row.type = IPROTO_FETCH_SNAPSHOT;
 	coio_write_xrow(io, &row);
 
-	applier_set_state(applier, APPLIER_FETCH_SNAPSHOT);
```

Comment:
There are two quite similar states, APPLIER_INITIAL_JOIN and APPLIER_FETCH_SNAPSHOT.

I remember asking you to delete the latter, because we don't wait for it anywhere.
You need to set the state to APPLIER_INITIAL_JOIN here instead, so that `applier_fetch_snapshot()` is tracked correctly in `replication.cc`.

Let's do this in a separate preparatory commit.

Reply:
@sergos, duplicate your comment please. I agree with it.

Reply:
> This commit doesn't seem make sense

Exactly. It only makes sense together with this PR. That's why Vlad asked you to send it as part of this PR.
Vlad also has 2 good points [here](https://github.com/tarantool/tarantool/pull/7183#discussion_r906427440) and [there](https://github.com/tarantool/tarantool/pull/7287#pullrequestreview-1014324471).

In summary:

    APPLIER_FETCH_SNAPSHOT -> APPLIER_WAIT_SNAPSHOT;
    APPLIER_CONTINUE_SNAPSHOT -> APPLIER_FETCH_SNAPSHOT;
    APPLIER_INITIAL_JOIN -> APPLIER_WAIT_SNAPSHOT;

And drop APPLIER_INITIAL_JOIN then.

I couldn't find any mentions of these states in doc or in tests, so we should be safe.